### PR TITLE
Add Rubocop Github Action job

### PR DESCRIPTION
**Pull request by lzap**
This is an attempt to move Rubocop from CI to GHAction

### CODE REVIEW

File: .github/workflows/rubocop.yaml
Comment (ekohl) [inline .github/workflows/rubocop.yaml:17]: https://github.com/andrewmcodes/rubocop-linter-action#rubocop-linter-action states:
> PLEASE DO NOT USE THIS ACTION, and instead follow [the Rails way](https://github.com/rails/rails/blob/4a78dcb326e57b5035c4df322e31875bfa74ae23/.github/workflows/rubocop.yml#L1).

Comment (lzap) [reply .github/workflows/rubocop.yaml]: Interesting, adopted Rails job. It's unfortunate not to see those nice comments but this will deliver a good experience too. It's super fast at least.

File: Gemfile
Comment (lzap) [inline Gemfile:18]: This is no longer in our template, honestly I don't know what is this and if this ever worked. There is no Rakefile in this repo so the instructions are not correct.

https://github.com/theforeman/foreman_plugin_template/blob/master/Gemfile

File: .github/workflows/rubocop.yaml
Comment (ekohl) [reply .github/workflows/rubocop.yaml]: The comments don't work on PRs from other repos, which is in practice all PRs.

**Comment by lzap** [comment]
@ekohl is this acceptable? Can I proceed with deleting the rake task call on our CI? I wonder if I can do this per plugin repo.

**Comment by ekohl** [comment]
We can run Rubocop in Jenkins as well and solve it for every plugin that uses Jenkins, but since there was basically no plugin maintainer response on my RFC I've concluded that plugin maintainers don't want any central support so I'm fine with everyone taking care of their own things.

**Comment by lzap** [comment]
Is there any change required in foreman-infra? I am trying to undrstand how all of this works, I think the script named `puppet/modules/jenkins_job_builder/files/theforeman.org/scripts/test/test_plugin.sh` is the correct one to do however I don't see anything related to rubocop there. And Foreman core`Rakefile` appears to be empty, weird.

**Comment by ekohl** [comment]
That's because plugins now typically add their Rubocop to the `jenkins:unit` rake task. So for this plugin it's here:
https://github.com/theforeman/foreman_bootdisk/blob/8ab8807715e2377f2c969a4b8f1740edc52827a4/lib/tasks/bootdisk.rake#L110

**Comment by lzap** [comment]
Ok I have removed this from bootdisk then, my understanding is that this is causing issues to Proxmox.

**Comment by ekohl** [comment]
> my understanding is that this is causing issues to Proxmox.

No, the issue in https://github.com/theforeman/foreman_bootdisk/pull/90 is that **foreman_fog_proxmox** needs to be modified. Removing it here will only help plugins that depend on foreman_bootdisk.

File: .github/workflows/rubocop.yaml
Comment (ezr-ondrej) [inline .github/workflows/rubocop.yaml:16]:
```suggestion
      uses: actions/cache@v2
```
V1 won't work for us (with the pattern `restore-keys`)

**Comment by lzap** [comment]
Let's hold this PR until @ezr-ondrej is done with core Rubocop changes and ability to share rules and then we can move from there.

File: .github/workflows/rubocop.yaml
Comment (ezr-ondrej) [inline .github/workflows/rubocop.yaml:14]: Confident much? xD
I feel it's pretty new ruby and foreman is not ready at all :thinking:

File: .github/workflows/rubocop.yaml
Comment (ezr-ondrej) [inline .github/workflows/rubocop.yaml:3]:
```suggestion
on: [pull_request]
```

**Review by ezr-ondrej — APPROVED** [review]
Few nits, loving it in general :+1:

File: .github/workflows/rubocop.yaml
Comment (lzap) [reply .github/workflows/rubocop.yaml:14]: I am running my dev setup on 2.7 for quite some time now. Granted I see a lot of warnings but 2.7.2 dropped these I think.

**Comment by lzap** [comment]
Ok rebased. I implemented `all.yaml` from the shared gem with some tuning :-)

**Comment by sbernhard** [comment]
> Let's hold this PR until @ezr-ondrej is done with core Rubocop changes and ability to share rules and then we can move from there.

Is there a link to the planned rubocop change?

**Comment by lzap** [comment]
This is ready for re-review, let's finally merge it and unblock ATIX guys with that proxmox PR?

**Comment by lzap** [comment]
> Is there a link to the planned rubocop change?

It is still WIP but I think we are good to proceed here. I am going to rebase and fix remaining problems.

https://community.theforeman.org/t/rubocop-common-styles-in-gem/20794

**Comment by lzap** [comment]
Success, this is now using pre-release of the foreman-rubocop which also supports github commenting.

**Comment by lzap** [comment]
@sbernhard would you be find with being nominated for bootdisk push access? You are the most active engineer in the last year and @timogoebel is busy so another member with push access would be great.

**Review by ezr-ondrej — APPROVED** [review]
Looks great @lzap !

**Comment by lzap** [comment]
@timogoebel would you mind?

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Thanks!

**Comment by sbernhard** [comment]
> @sbernhard would you be find with being nominated for bootdisk push access? You are the most active engineer in the last year and @timogoebel is busy so another member with push access would be great.

sure! I would do my best :)

**Comment by lzap** [comment]
https://community.theforeman.org/t/nominating-bernhard-suttner-for-push-access-to-foreman-bootdisk/23144

**Pull request merged**
